### DISCUSSION TITLE
Is there a way to scope Volar to only work on a single package in a monorepo?

### DISCUSSION THREAD

**Discussion by fabis94** [vuejs/language-tools#1797] [Q&A]
Question 1:
I'm a bit concerned by Volar's resource usage, so I want to ensure that only the Vue app package in the monorepo runs the Vue/TS language server, and all of the other packages rely on the default/standard TS language server.

Is there a way to configure this? Or is Volar smart enough to only "turn on" and scan frontend projects that have Vue in them?

Question 2:
I see the "take over mode" being advertised as a performance optimization for Volar, but it's no longer referenced on the Volar VSCode extension page, so I'm wondering - is this still something devs are supposed to be doing? I've never got this mode working well, the entire TS experience was always somehow better or at the very least faster with this plugin instead: https://github.com/johnsoncodehk/volar/tree/master/extensions/vscode-typescript-vue-plugin

Question 3:
How do you enable "take over mode" for a single package in a monorepo? The instructions disable the built-in TS server for the entire monorepo

**Comment by johnsoncodehk** [comment]
Question 1:
When you're not using takeover mode, language server only create language service instance when you open a vue file, so yes only TS project language services that included vue files will be create. https://github.com/johnsoncodehk/volar/blob/2758510a8e519563f16ce023c700ab98b8b65ad8/packages/vue-language-server/src/utils/workspaceProjects.ts#L84-L93

Question 2:
For example, when you developing in https://github.com/johnsoncodehk/volar-starter, there have 1 tsconfig.json (Each tsconfig have it own language service instance in language server), with "TypeScript Vue Plugin" we will have 3~5 TS language service instances:

- 1 / 2 from tsserver of VSCode built-in TypeScript extension
- 1 / 2 from "TypeScript Vue Plugin" for use in *.ts
- 1 from "Vue Language Features" for use in *.vue

"TypeScript Vue Plugin" and "Vue Language Features" language service instances is not sharing it internal cache, so when you switch coding from .ts to .vue (or from .vue to .ts), you should feel a moment slow because the language service that now usage have to catch up all changes since you last used it. If you keeping switch coding in .vue and .ts this situation will keep happening.

With takeover mode it only have 1 TS langauge service instance.

- 1 from "Vue Language Features" for use in *.vue and *.ts

So there are less memory usage and avoid multiple language service cache cannot share issue.

You can try enable "volar.vueserver.useSecondServer" in VSCode to have faster intellisense, it make "Vue Language Features" use 2 TS language services for each tsconfig just like tsserver.

Question 3:
You can't, but nice to have a feature request.
